lib/ansible/module_utils/redfish_utils.py

```diff
@@ -0,0 +1,932 @@
+        response = self.get_request(self.root_uri + storage_uri)
+        if response['ret'] is False:
+            return response
+        result['ret'] = True
+        data = response['data']
+
+        for controller in data[u'Members']:
+            controller_list.append(controller[u'@odata.id'])
+
+        for c in controller_list:
+            uri = self.root_uri + c
+            response = self.get_request(uri)
+            if response['ret'] is False:
+                return response
+            data = response['data']
+
+            for device in data[u'Devices']:
+                disk = {}
+                disk['Controller'] = data[u'Name']  # Name of storage controller
+                disk['Name'] = device[u'Name']
+                disk['Manufacturer'] = device[u'Manufacturer']
+                disk['Model'] = device[u'Model']
+                disk['State'] = device[u'Status'][u'State']
+                disk['Health'] = device[u'Status'][u'Health']
+                disks_details.append(disk)
+
+        result["entries"] = disks_details
+        return result
+
+    def restart_manager_gracefully(self):
+        result = {}
+        key = "Actions"
+
+        # Search for 'key' entry and extract URI from it
+        response = self.get_request(self.root_uri + self.manager_uri)
+        if response['ret'] is False:
+            return response
+        result['ret'] = True
+        data = response['data']
+        action_uri = data[key]["#Manager.Reset"]["target"]
+
+        payload = {'ResetType': 'GracefulRestart'}
+        response = self.post_request(self.root_uri + action_uri, payload, HEADERS)
+        if response['ret'] is False:
+            return response
+        return {'ret': True}
+
+    def manage_system_power(self, command):
+        result = {}
+        key = "Actions"
+
+        # Search for 'key' entry and extract URI from it
+        response = self.get_request(self.root_uri + self.systems_uri)
+        if response['ret'] is False:
+            return response
+        result['ret'] = True
+        data = response['data']
+        action_uri = data[key]["#ComputerSystem.Reset"]["target"]
+
+        # Define payload accordingly
+        if command == "PowerOn":
+            payload = {'ResetType': 'On'}
+        elif command == "PowerForceOff":
+            payload = {'ResetType': 'ForceOff'}
+        elif command == "PowerGracefulRestart":
+            payload = {'ResetType': 'GracefulRestart'}
+        elif command == "PowerGracefulShutdown":
+            payload = {'ResetType': 'GracefulShutdown'}
+        else:
+            return {'ret': False, 'msg': 'Invalid Command'}
+
+        response = self.post_request(self.root_uri + action_uri, payload, HEADERS)
+        if response['ret'] is False:
+            return response
+        result['ret'] = True
+        return result
+
+    def list_users(self):
+        # listing all users has always been slower than other operations, why?
+        result = {}
```

Comment:
Should this be `response` since that is what is returned? I don't see `result` used within this function.